File: 2020/2/2dec.hs
-- import System.IO
-- import Data.List

-- -- getMin :: String -> Int
-- -- getMin input = read mn input

-- -- mn :: String -> String
-- -- mn (c:rest)
-- --     | c == '-'  = ""
-- --     | otherwise = c : mn rest

-- -- getMax :: String -> Int
-- -- getMax input = read [input !! 2] :: Int

-- -- mx :: String -> String
-- -- mx (c:rest)
-- --     | c == ' '  = ""


-- getLetter :: String -> Char
-- getLetter input = input !! 4

-- getOccurences :: String -> Char -> Int
-- getOccurences input c = sum [1 | a <- input, a == c]

-- getPsw :: String -> String
-- getPsw (c:rest)
--     | c == ':'  = rest
--     | otherwise = getPsw rest

-- getMax input = elemIndex '-' input

-- main :: IO ()
-- main = do
--     handle <- openFile "2dec.txt" ReadMode
--     input <- hGetContents handle
--     let valid = length [1 | psw <- lines input, getOccurences (getPsw input) (getLetter psw) >= getMin psw, getOccurences (getPsw input) (getLetter psw) <= getMax psw]
--     print valid
--     hClose handle 
    

File: 2020/2/2dec.py


with open("2dec.txt") as f:
    inputs = f.readlines()


def part1(inputs):
    count = 0
    for psw in inputs:
        mn, rest = psw.split("-")
        mx, c, passw = rest.split(" ")
        c = c[0]
        mn = int(mn)
        mx = int(mx)
        if passw.count(c) <= mx and passw.count(c) >= mn:
            count += 1

        # print("c:", c, ", mn:", mn, ", mx:", mx, ", pass:", passw)

    return count
    
def part2(inputs):
    count = 0
    for psw in inputs:
        mn, rest = psw.split("-")
        mx, c, passw = rest.split(" ")
        c = c[0]
        mn = int(mn)
        mx = int(mx)
        if passw[mn-1] == c and passw[mx-1] != c:
            count += 1
        elif passw[mn-1] != c and passw[mx-1] == c:
            count += 1

        # print("c:", c, ", mn:", mn, ", mx:", mx, ", pass:", passw)

    return count

print("Part1:", part1(inputs))
print("Part2:", part2(inputs))